torch/_dynamo/variables/tensor.py

```diff
@@ -644,6 +644,46 @@ def redistribute_fn_with_prim_types(x):
                 ),
                 **options,
             )
+        elif name == "register_hook":
+            # see [On tensor.register_hook]
+            assert len(args) == 1
+            fn_var = args[0]
+
+            if isinstance(fn_var, variables.NestedUserFunctionVariable):
+                # NestedUserFunctionVariable don't carry their fn, but reconstruction builds it
+                # This should not be onerous to support when needed.
+                unimplemented("NYI - lambda variables as hooks")
+            elif isinstance(fn_var, variables.functions.FunctoolsPartialVariable):
+                fn = fn_var.as_python_constant()
+                name = fn_var.func.fn.__name__
+            else:
+                fn = fn_var.fn
+                name = fn_var.fn.__name__
+
+            handle = self.as_proxy().node.meta["example_value"].register_hook(fn)
```

Comment:
tbd if we need this

Reply:
Yea we do, its for handle.remove() subsumed by compile

Reply:
No, I don't understand. Why do you need the handle?

Reply:
Nah, I was wrong!!